modules/dataplex-autodq/main.tf

```diff
@@ -0,0 +1,187 @@
+/**
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+locals {
+  prefix                    = var.prefix == null ? "" : "${var.prefix}-"
+  _parsed_data_quality_spec = try(yamldecode(file(var.data_quality_spec_file.path)), null)
+  _parsed_data_quality_spec_convert_camel_case = (
+    try(var.data_quality_spec_file.convert_camel_case, false) ?
+    {
+      sampling_percent = try(local._parsed_data_quality_spec.samplingPercent, null)
+      row_filter       = try(local._parsed_data_quality_spec.rowFilter, null)
+      rules = [
+        for rule in local._parsed_data_quality_spec.rules : {
+          for k, v in rule :
+          join("_", [for word in flatten(regexall("((?:[A-Z]|[0-9]+)[a-z]*)", title(k))) : lower(word)])
+          => try({
+            for kk, vv in v :
+            join("_", [for word in flatten(regexall("((?:[A-Z]|[0-9]+)[a-z]*)", title(kk))) : lower(word)])
+            => vv
+          }, v)
+        }
+      ]
+    } :
+  null)
```

Comment:
the else part should be `local._parsed_data_quality_spec` no?

Reply:
This has been refactored away.